testsuite/tests/lib-threads/sockets.ml

```diff
@@ -26,13 +26,18 @@ let server sock =
     ignore(Thread.create serve_connection s)
   done
 
-let client (addr, msg) =
+let client_turn = ref 1
+
+let client (id, addr) =
+  let msg = "Client #" ^ Int.to_string id ^ "\n" in
   let sock =
     Unix.socket (Unix.domain_of_sockaddr addr) Unix.SOCK_STREAM 0 in
   Unix.connect sock addr;
   let buf = Bytes.make 1024 ' ' in
   ignore(Unix.write_substring sock msg 0 (String.length msg));
   let n = Unix.read sock buf 0 (Bytes.length buf) in
+  while not (!client_turn = id) do Thread.yield () done;
```

Comment:
Note to self: this looks like the busy-wait pattern that we are taught to avoid, but in fact it is okay because this function itself is the one that makes progress by incrementing `client_turn` below (unlocking another thread), so calling `Thread.yield ()` is not busy-waiting but actually progressing towards termination.

Reply:
Yes. I won’t claim this is a very efficient way of doing it as the thread might be woken up one or more times with the loop condition still not fulfilled; using a mutex and a condition variable would probably be more efficient, but for such a simple we went for the first simple solution that seemed correct.

Reply:
Personally I would feel more confident that the test behaves correctly with a condition variable rather than a busy-wait loop, that looks tricky to think about and more likely to backfire.

Reply:
Following your idea to not to rely on an explicit busy loop, Olivier and I reimplemented the synchronisation with an `Event.channel` which internally uses `Mutex` and `Condition`.